node/src/Commands.ts

```diff
@@ -804,27 +804,133 @@ export function createZscore(
 }
 
 export type ScoreLimit =
+    /**
+     * Positive infinity bound for sorted set.
+     */
     | `positiveInfinity`
+    /**
+     * Negative infinity bound for sorted set.
+     */
     | `negativeInfinity`
+    /**
+     *  Represents a specific numeric score boundary in a sorted set.
+     */
     | {
-          bound: number;
+          /**
+           * The score value.
+           */
+          value: number | string;
+          /**
+           * Whether the score value is inclusive. Defaults to True.
+           */
           isInclusive?: boolean;
       };
 
-function getScoreLimitArg(score: ScoreLimit): string {
+/**
+ * Represents a range by index (rank) in a sorted set.
+ * The `start` and `stop` arguments represent zero-based indexes.
+ */
+export type RangeByIndex = {
+    /**
+     *  The start index of the range.
+     */
+    start: number;
+    /**
+     * The stop index of the range.
+     */
+    stop: number;
+};
+
+/**
+ * Represents a range by score or a range by lex in a sorted set.
+ * The `start` and `stop` arguments represent score boundaries.
+ */
+export type RangeByScoreOrLex = {
```

Comment:
Why not using two types as we do in python, RangeByScore and RangeByLex? we shouldn't accept ScoreLimit as strings when is lex is false, and we should except number when is lex is true. We also cannot use it later on with the specific zrange functions (e.g. ZRANGEBYLEX). Please follow python's design